src/main/java/io/tiledb/java/api/Array.java

```diff
@@ -615,6 +615,151 @@ public Pair getNonEmptyDomainFromName(String name) throws TileDBError {
     return ret;
   }
 
+  /**
+   * Get a metadata key-value item from an open array. The array must be opened in READ mode,
+   * otherwise the function will error out.
+   *
+   * @param key a key to retrieve from the metadata key-value
+   * @param nativeType The Datatype
+   * @return NativeArray which contains the metadata
+   * @throws TileDBError A TileDB exception
+   */
+  public NativeArray getMetadata(String key, Datatype nativeType) throws TileDBError {
+    checkIsOpen();
+
+    SWIGTYPE_p_p_void resultArrpp = tiledb.new_voidpArray(0);
+    SWIGTYPE_p_unsigned_int value_num = tiledb.new_uintp();
+    SWIGTYPE_p_tiledb_datatype_t value_type =
+        tiledb.copy_tiledb_datatype_tp(nativeType.toSwigEnum());
```

Comment:
I thought that [nativeType is not required](https://github.com/TileDB-Inc/TileDB-Go/blob/3479a6c12c1939a44aa84b09f5140aa3cc3a4ab2/array.go#L1107-L1114) for the input.
However, `tiledb.tiledb_array_get_metadata` would throw a null pointer exception in case `value_type = tiledb.new_tiledb_datatype_tp();`